alma/controllers/hook/displayPayment.php

```diff
@@ -80,9 +80,15 @@ public function run($params)
                 }
             }
 
+            $alma = AlmaClient::defaultInstance();
+            $paymentDataPnX = PaymentData::dataFromCart($this->context->cart, $this->context, $n);
+            $eligibilityPnX = $alma->payments->eligibility($paymentDataPnX);  
```

Comment:
I think we need some heavy refactoring here:

- `AlmaEligibilityHelper::eligibilityCheck` should be the one to loop on `$n` and check eligibility for each activated installments count:
    - Get plans activated by the merchant
    - Check amount bounds for each activated plans: skip those for which amount is outside of bounds
    - Call Alma eligibility API for activated plans that should be active for cart amount
    - Return eligibility results

Something like this:
```
$eligibilities = [];
$activePlans = [];
foreach (AlmaSettings::activatedInstallmentsPlans() as $n) { // [2,3,4]
	if ($purchaseAmount < AlmaSettings::installmentPlanMinAmount($n) || $purchaseAmount > AlmaSettings::installmentPlanMaxAmount($n)) {
		$eligibility = new stdClass();
		$eligibility->installments_count = $n;
		$eligibility->isEligible = false;
		$eligibility->constraints = array(
			"purchase_amount" => array(
				"minimum" => AlmaSettings::installmentPlanMinAmount($n),
				"maximum" => AlmaSettings::installmentPlanMaxAmount($n)
			)
		);

		$eligibilities[] = $eligibility;
	} else {
		$activePlans[] = $n;
	}
}

$paymentData = PaymentData::dataFromCart($context->cart, $context, $activePlans);
$almaEligibilities = $alma->payments->eligibility($paymentData);

$eligibilities = array_merge($eligibilities, $almaEligibilities);

function cmp_installments_count($a, $b) {
	return $a->installments_count > $b->installments_count ? 1 : ($a->installments_count == $b->installments_count ? 0 : -1);
}

usort($eligibilities, 'cmp_installments_count');
return $eligibilities;
```

- Loop on each eligibility result and and display only a button for the eligible ones

Reply:
Do check Shoping cart footer hook controller & paymentOptions hook controller to adapt them to the new eligibility check